### PR TITLE
[DSY-778] Docs/add contribution docs index

### PR DESCRIPTION
**Pull request by arielwb**
# Description

This prs improves the contribution documentation.
- Added storybook, code quality, pull request and publishing sections
- Split sections into files
- Add the `.md` docs to storybook-web
- Add templates for issues and pull requests

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change is a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,118 +5,9 @@
...
-
-
-feat(button): add button component
-
-co-author: @somebody
-DSY-766
-```
+- Make sure your development environment meets the [prerequisites](./docs/Prerequisites.md)
+- [Install](./docs/Installation.md) the project on your machine
+- Open a branch and [make your changes](./docs/MakingChanges.md)
```
Comment (handrus) [inline CONTRIBUTING.md:10]: Maybe something like:
Open a branch if you have permission, else [fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) this repository and [make your changes](./docs/MakingChanges.md)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -5,118 +5,9 @@
 Follow the steps bellow to add a new feature, make an improvement or fix a bug.
 (Make sure to read the guides on each step as they contain important additional information)
```
Comment (handrus) [inline CONTRIBUTING.md:7]: We should add a warning that the following process is fully tested only for internal users that have access to open a branch, and make use of our pipelines.Maybe reinforce we are open to issues and would like to improve this process to external users.

File: docs/Commiting.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+This project follows the [conventional commits](https://www.conventionalcommits.org) guidelines to automate our version management and changelog generation.
+A pre-commit hook runs [commit lint](https://commitlint.js.org/) before every commit to ensure commit consistency.
+If you need some help commiting, try running `yarn commit` to commit using the [commitizen](https://github.com/commitizen/cz-cli) tool.
+
+> Be sure to check out those links for more information on how conventional commits guidelines and tools
+
+## Commit requirements
+
+- A commit message should follow the conventional commits guidelines
+- A commit message should include an issue number (DSY-*)
```
Comment (handrus) [inline docs/Commiting.md:12]: We should also let people reference a issue on github issues.
If this is too much of a work... lets add a warning note, something like "Not yet ready for external contributions. If you'd like to contribute and you are not a developer on Natura organization, please open a issue for us. We'd like to discuss our contribution proccess"

File: docs/Commiting.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+### Commit Examples
+
+```sh
+docs: update readme, move contributing section to new file
+DSY-778
```
Comment (handrus) [inline docs/Commiting.md:18]: We should add a example for external users, referencing a github issue.

File: docs/PullRequests.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+- The design system team will start the review
+
+### Pipeline
+
+Our pull request pipeline will run the following steps:
+
+- code quality (tests, lint, type checking)
+- build and publish the `ios` sample app on the iTunes connect for review
+- build the `android` sample app and distribute via bitrise install page for review
+- build and publish a test version the storybook documentation on netlify for review (accessible by the url `natds-rn-your-branch-name.netlify.app`)
```
Comment (handrus) [inline docs/PullRequests.md:28]: Does it work if the PR is coming from a forked repo?
If not, we should just add a warning that this section (pipeline works only for contributors with access to this repository and doesn't apply for forked repositories

File: docs/Publishing.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+Merging your code means that the features or fix was reviewed and approved by the Design System Team and that a new version is ready to be published!
+When you merge you code you are all done here. A new version of the library will be release as soon as the pipeline finishes.
+
+## Before merging your code
+
+Before opening a merging, please be sure that:
+
+- your code was reviewed and approved by the Design System Team developers
+- your changes were visually validated by the Design System Team designers
+- your changes were approved and the jira issue was moved to waiting release
```
Comment (handrus) [inline docs/Publishing.md:12]: Add comment: "Doesn't apply if you are a developer outside Natura organization"

File: docs/PullRequests.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
+# Pull Requests
+
+## Before opening a pull request
+
+Before opening a pull request, please be sure that:
+
+- your code follows our [contributing guidelines](./ContributionGuidelines.md) and that your [code quality checks](./CodeQuality.md) pass locally.
+- you tested your code on an ios and android and made sure that both are consistent with our visual guidelines.
```
Comment (handrus) [inline docs/PullRequests.md:8]: Is there a link to our visual guidelines?

File: docs/PullRequests.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+- you documented your code. If it is a component, make sure you documented the props and added examples in you sample app and web documentation.
```
Comment (handrus) [inline docs/PullRequests.md:9]: Should we add something about storybook? Like:

- make sure your documentation is rendering and legible on storybook

Comment (arielwb) [reply docs/PullRequests.md]: boa! done in c5e361b4c7b94fd71b8f4c1d35d2f3b453a4bbc3

File: docs/PullRequests.md (hunk 7 above)
Comment (arielwb) [reply docs/PullRequests.md]: adicionei um link pro zero height ;) 
done in c5e361b

File: docs/Publishing.md (hunk 6 above)
Comment (arielwb) [reply docs/Publishing.md:12]: boa! done in c5e361b

File: docs/PullRequests.md (hunk 5 above)
Comment (arielwb) [reply docs/PullRequests.md:28]: It will not work from a fork, I added a note
done in c5e361b

File: docs/Commiting.md (hunk 4 above)
Comment (arielwb) [reply docs/Commiting.md:21]: Não adicionei um exemplo porque hj eu dependo disso, tem que ajustar uma coisita no codigo pra aceitar esse padrão tbm. adicionei uma nota por enquanto e vou criar um card no jira so pra gente deixar o repo e processos amigaveis pra contribuições externas ;)

done in c5e361b

File: docs/Commiting.md (hunk 3 above)
Comment (arielwb) [reply docs/Commiting.md]: Adicionei a nota, done in c5e361b

File: CONTRIBUTING.md (hunk 2 above)
Comment (arielwb) [reply CONTRIBUTING.md:7]: Inclui nesse arquivo um "disclaimer" que o nosso processo de contribuição ta preparado somente pra contribuições internas no momento, e que pretendemos melhorar isso no futuro ;)
done in c5e361b

File: CONTRIBUTING.md (hunk 1 above)
Comment (arielwb) [reply CONTRIBUTING.md]: boa! done in c5e361b

**Comment by handrus** [comment]
LGTM

**Pull request merged**